ci-operator/step-registry/ipi/conf/azure/resourcegroup/ipi-conf-azure-resourcegroup-commands.sh

```diff
@@ -10,24 +10,30 @@ curl -L https://github.com/mikefarah/yq/releases/download/3.3.0/yq_linux_amd64 -
 CONFIG="${SHARED_DIR}/install-config.yaml"
 PATCH="/tmp/install-config-existingresourcegroup.yaml.patch"
 
-azure_region=$(/tmp/yq r "${CONFIG}" 'platform.azure.region')
-cluster_name=$(/tmp/yq r "${CONFIG}" 'metadata.name')
-existing_rg=${cluster_name}-exrg
-
-# az should already be there
-command -v az
-
-# set the parameters we'll need as env vars
-AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
-AZURE_AUTH_CLIENT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientId)"
-AZURE_AUTH_CLIENT_SECRET="$(<"${AZURE_AUTH_LOCATION}" jq -r .clientSecret)"
-AZURE_AUTH_TENANT_ID="$(<"${AZURE_AUTH_LOCATION}" jq -r .tenantId)"
-
-# log in with az
-az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none
-
-# create resource group prior to installation
-az group create -l "${azure_region}" -n "${existing_rg}"
+provisioned_rg_file="${SHARED_DIR}/resouregroup"
+if [ -f "${provisioned_rg_file}" ]; then
```

Comment:
It does not seem worth reusing this file. In general we discourage this kind of conditionality in steps. Because steps is design to allow you to pick and choose the individual steps for your workflow, there shouldn't be as much need to create conditional paths.

Instead of reusing this step just to get the patch for the resource group,  could you rework `ipi-conf-azure-provisionnetwork` so that step looks for every provisioned resource and then patches the install config directly?

Reply:
Per my understanding, "customer provisioned network" and "customer provisioned resource group" are two difference cases, they are not have to be combined together (in this special case, we need both).
So sounds like we need to introduce another new step (maybe call it `ipi-conf-azure-provisioned-resourcegroup`) to avoid reusing this file?

Reply:
Yes, I would prefer that we do not reuse this file. 

You could create a specific step to patch files for this job, but let me clarify my suggestion about the `ipi-conf-azure-provisionnetwork` step. My suggestion is you change that step to something like `ipi-conf-patch` and use it in both the customer provisioned network and resource group flows. The install config would be patched differently for the network case vs the resource group case based on environment variables. 

I am also fine with just creating a dedicated patch step for this job.

Reply:
> My suggestion is you change that step to something like ipi-conf-patch and use it in both the customer provisioned network and resource group flows. The install config would be patched differently for the network case vs the resource group case based on environment variables.

Got your points. Once that, will also need some judgement conditions for environment variables in the `ipi-conf-patch` to do different patch operations, I prefer to create a dedicated `ipi-conf-azure-provisioned-resourcegroup` to do the dedicated patch operation, so that users can choose the individual steps (or both) in his/her workflow.